Dockerfile

```diff
@@ -50,19 +54,41 @@ RUN . /opt/ros/melodic/setup.bash && \
 RUN echo "source ~/catkin_ws/devel/setup.bash" >> ~/.bashrc && \
     source ~/.bashrc
 
-# TODO(ycho):
-# sudo apt install ros-melodic-libpcan # required for building with robotnik_base_hw_lib
+# required for building with robotnik_base_hw_lib
+# RUN sudo apt-get install ros-melodic-libpcan
 # RUN sudo dpkg -i ros-melodic-robotnik-base-hw-lib_1.0.0-0bionic_amd64.deb ros-melodic-robotnik-msgs_1.0.0-0bionic_amd64.deb 
-# whitelist libbw_ce30v2.0.so from .gitignore (for now, replace with fork OR self-hosted .zip from official repo)
-# rosdep install libpcan
-# rosmake libpcan
-# OR, add <build_depend>libpcan</build_depend> in robotnik*/package.xml
-# and CATKIN_FIND_PACKAGE(... libpcan)
-# and INCLUDE_DIRECTORIES(... "/opt/ros/melodic/include/libpcan/")
-# some combination of the above works. My guess is rosdep/rosmake doesn't do much.
+RUN rosdep update
+
+# NOTE(ycho): --mount=type=ssh,uid=${UID} doesn't work
+# Tracked Issue: https://github.com/moby/buildkit/issues/815
+# Until then, we need to clone as root, chown then copy the resulting directory to workspace.
+# When resolved, use the following instead of the workaround:
+# RUN --mount=type=ssh mkdir -p -m 0600 "~/.ssh" \
+#     && ssh-keyscan github.com | sudo tee -a "~/.ssh/known_hosts"
+# RUN --mount=type=ssh git clone git@github.com:iMSquared/imm-summit-packages.git --depth 1 -b full \
+#     "~/catkin_ws/src/imm-summit-packages/"
+RUN sudo chown ${USERNAME} -R /tmp/imm-summit-packages && \
+    mv /tmp/imm-summit-packages ~/catkin_ws/src/
 
-# Clone our workspace and build.
-# RUN pushd src && git clone imsquared/imm-summit-packages && popd
-# RUN sudo apt update
+# Build ...
 # RUN rosdep install --from-paths src --ignore-src -y --skip-keys='robotnik_base_hw_lib' --skip-keys='robotnik_pose_filter' --skip-keys='robotnik_locator'
-# RUN catkin_make
+RUN sudo apt-get update && sudo apt-get install -y \
+    ros-melodic-libpcan
+
+# NOTE(ycho): Not directly running install_debs.sh since it doesn't include proper cli flags.
+# FIXME(ycho): Monkey-patched gazebo dependencies, will not work!!!
+RUN sudo apt-get install -y libgazebo9-dev ros-melodic-gazebo-dev
+RUN cd ~/catkin_ws/src/imm-summit-packages/deb && \
+    sudo apt-get update && \
+    sudo apt-get install -y \
+    $(cat ./install_debs.sh | grep deb$ | awk '{print "./"$4}' | grep -v 'melodic-gazebo') \
+    && rm -rf /var/lib.apt/lists/*
+
+# RUN rosdep install libpcan && rosmake libpcan
+RUN cd ~/catkin_ws && rosdep install --from-paths src --ignore-src -y \
+    --skip-keys='robotnik_base_hw_lib' \
+    --skip-keys='robotnik_pose_filter' \
+    --skip-keys='robotnik_locator'
+RUN . /opt/ros/melodic/setup.bash && \
+    cd ~/catkin_ws && \
+    catkin_make
```

Comment:
The last line (catkin_make) fails... Please see the attached log file.
[build.log](https://github.com/iMSquared/imm-docker-ros/files/6935559/build.log)

Reply:
Interesting. Looking into it with a fresh build with no cache.

Reply:
@Minchan-Kim Updated and tested build with and without cache, appears to be OK now.
Not sure why the problem did not occur on the other setup; for now, I'm assuming it was a cache problem.
Let me know how the build goes for you.

Reply:
New commit resolved the problem. I built the image successfully. Thank you!

Reply:
@Minchan-Kim Just so that we get in the habit of it, can you `Approve` the PR if the image works?